packages/remix-express/server.ts

```diff
@@ -38,14 +50,52 @@ export type RequestHandler = (
 export function createRequestHandler({
   build,
   getLoadContext,
+  serverMiddleware,
   mode = process.env.NODE_ENV,
 }: {
   build: ServerBuild;
   getLoadContext?: GetLoadContextFunction;
+  serverMiddleware?: ServerMiddlewareFunction;
   mode?: string;
 }): RequestHandler {
   let handleRequest = createRemixRequestHandler(build, mode);
 
+  if (build.future.unstable_middleware && serverMiddleware) {
+    return async (
+      req: express.Request,
+      res: express.Response,
+      next: express.NextFunction
+    ) => {
+      let response: NodeResponse | undefined;
+      let context = createMiddlewareStore();
```

Comment:
It's worth noting that at the moment, since our context is keyed by object instances for strong-typing, users can run into cross-boundary CJS/ESM instance issues when using `serverMiddleware`.  This is solveable by either (1) bundling your entire server via the `remix.config.js` `server` option, or (2) potentially using import maps in your application `package.json` (untested)